### PR TITLE
[Task05]

### PR DESCRIPTION
**Pull request by ando-andy**
## Related Tickets

- [#Ticket ID](https://docs.google.com/spreadsheets/d/1F5V2sIUMGej3Wrp8qYRscf0KaVSlccBmqkRx49_hUVo/edit#gid=935273790)

## What's this PR do ?
- Search screen created

## Library
*(List gem, library third party add new)*
https://developer.apple.com/documentation/uikit/uisearchcontroller
https://github.com/tuan188/MGCleanArchitecture

## Impacted Areas in Application
*(List features, api, models or services that this PR will affect)*

## Checklist

- [x] It was tested in local success ?
- [x] Updated rake task, environment variable at [here](https://???)
- [x] Fill link PR into ticket and the opposite
- [x] Note reason, scope of influence, solution into ticket
- [x] Validate UI/Model/API

## Notes
*(Other notes)*

### CODE REVIEW

File: MovieDB/Scenes/Search/SearchViewController.swift [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+// MARK: - Delegaiton
+extension SearchViewController: UISearchBarDelegate {
+    
+    func searchBarTextDidBeginEditing(_ searchBar: UISearchBar) {
+        searchWasCancelled = false
+    }
+    
+    func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
+        searchTerms = searchBar.text ?? " "
```
Comment (longvt-1468) [inline MovieDB/Scenes/Search/SearchViewController.swift]:
```suggestion
        searchTerms = searchBar.text ?? ""
```

File: MovieDB/Scenes/Search/SearchViewController.swift [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+final class SearchViewController: UIViewController, BindableType {
+    
+    // MARK: - IBOutlets
+    
+    @IBOutlet weak var tableView: RefreshTableView!
+    
+    // MARK: - Properties
+    
+    var viewModel: SearchViewModel!
+    var searchTerms = ""
```
Comment (longvt-1468) [inline MovieDB/Scenes/Search/SearchViewController.swift]: add private

File: MovieDB/Scenes/Search/SearchViewController.swift [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    var searchWasCancelled = false
```
Comment (longvt-1468) [inline MovieDB/Scenes/Search/SearchViewController.swift]: add private

File: MovieDB/Scenes/Search/SearchViewController.swift [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    
+    // MARK: - Methods
+
+    private func configView() {
+        self.do {
+            $0.definesPresentationContext = true
+            $0.title = "Search Movie"
+        }
+        
+        self.searchController.do {
```
Comment (longvt-1468) [inline MovieDB/Scenes/Search/SearchViewController.swift]: remove self

File: MovieDB/Scenes/Search/SearchViewController.swift [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+            $0.searchBar.delegate = self
+            $0.searchBar.placeholder = "Enter Keyword"
+        }
+        
+        self.navigationItem.do {
```
Comment (longvt-1468) [inline MovieDB/Scenes/Search/SearchViewController.swift]: remove self

File: MovieDB/Scenes/Search/SearchViewController.swift [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+    func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
+        searchTerms = searchBar.text ?? " "
+        searchWasCancelled = true
+    }
+    
+    func searchBarTextDidEndEditing(_ searchBar: UISearchBar) {
+        if searchWasCancelled {
+            searchBar.text = self.searchTerms
+        } else {
+            searchTerms = searchBar.text ?? " "
```
Comment (longvt-1468) [inline MovieDB/Scenes/Search/SearchViewController.swift]:
```suggestion
            searchTerms = searchBar.text ?? ""
```

File: MovieDB/Scenes/Search/SearchViewController.swift (hunk 1 above)
Comment (ando-andy) [reply MovieDB/Scenes/Search/SearchViewController.swift]: Fixed 🙇‍♂️

File: MovieDB/Scenes/Search/SearchViewController.swift (hunk 2 above)
Comment (ando-andy) [reply MovieDB/Scenes/Search/SearchViewController.swift]: Fixed 🙇‍♂️

File: MovieDB/Scenes/Search/SearchViewController.swift (hunk 3 above)
Comment (ando-andy) [reply MovieDB/Scenes/Search/SearchViewController.swift]: Added 🙇‍♂️

File: MovieDB/Scenes/Search/SearchViewController.swift (hunk 4 above)
Comment (ando-andy) [reply MovieDB/Scenes/Search/SearchViewController.swift]: Removed 🙇‍♂️

File: MovieDB/Scenes/Search/SearchViewController.swift (hunk 5 above)
Comment (ando-andy) [reply MovieDB/Scenes/Search/SearchViewController.swift]: Removed 🙇‍♂️

File: MovieDB/Scenes/Search/SearchViewController.swift (hunk 6 above)
Comment (ando-andy) [reply MovieDB/Scenes/Search/SearchViewController.swift]: Fixed 🙇‍♂️

**Pull request merged**